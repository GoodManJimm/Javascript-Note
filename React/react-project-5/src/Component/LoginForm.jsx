import React, { useEffect } from 'react';
// import useDocumentClick from './utils/hooks/useDocumentClick';

export default function LoginForm()
{
    useEffect(() => 
        {
          const resizeEventHandler = (e) => 
          {
            console.log("Window/viewpoint Resized")
          }


          window.addEventListener('resize',resizeEventHandler)
    
          return () => {
              window.removeEventListener('resize',resizeEventHandler)
              console.log("Removing Resize event listener");
              console.log("unmount login form");
            }
        },[]);
    
        // useDocumentClick()
        
        return (
            <form  onSubmit={(e) => 
            {
                e.preventDefault()
                const formData = new FormData(e.target)
                console.log(formData);
                console.log(formData.get("username"));
                const username = formData.get("username")
                const password = formData.get("password")
                // fetch("http://localhost:3001/api/login",{body: 
                // {
                //     username,
                //     password,
                // },
                // method:"POST"})
            }}>
                <label htmlFor="username">Username: </label>
                <br />
                <input type="text" id='username' name='username'
                onChange={(e) => {
                    console.log("Value changed")
                    console.log(e.target);
                    console.log('Username: ' + e.target.value)
    
                }}  onKeyDown={(e) => 
                {
                    console.log(e.key);
                }}/>
                <br />
                <label htmlFor="password">Password: </label>
                <br />
                <input type="password" name="password" id="password" onChange={(e) => {
                    console.log("Value changed")
                    console.log(e.target);
                    console.log('Password: ' + e.target.value)
    
                }} />
                <br />
                <button>Login</button>
                <br />
            </form>
        )
    
}