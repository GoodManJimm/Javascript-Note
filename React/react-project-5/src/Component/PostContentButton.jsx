import { useContext, useState } from "react"
import { UserContext } from "../Context/UserContext"



export default function PostContentButton()
{
    const {id,setUserData} = useContext(UserContext)
    const [value, setValue] = useState("")

    return (
        <div>
            <div>
                <span>PostContentButton</span>
            </div>
            {id}
            <br />
            <label htmlFor="UpdateName">Update Name:</label>
            <input id="UpdateName"type="text" value={value} onChange={(e) => {
                setValue(e.target.value)
            }}/>
            <button onClick={(e) => 
                {
                    setUserData((currentState) => (
                        {
                            ...currentState, name: value
                        }))
                }}>Update display name</button>
        </div>
    )
}