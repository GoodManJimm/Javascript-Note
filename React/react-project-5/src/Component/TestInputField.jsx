import { input } from "@testing-library/user-event/dist/cjs/event/input.js";

export function TestInputField(){
    return( 
        <input placeholder="enter data..." value="hello" />
    )
}