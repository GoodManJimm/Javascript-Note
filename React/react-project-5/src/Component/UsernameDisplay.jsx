import { useEffect, useState } from "react"

export function UsernameDisplay({username})
{

    const [timePassed, setTimePassed] = useState(false)

    useEffect(() => {
        setTimeout(() => {
            setTimePassed(true)
        },1000)
    },[])
    return(
        <div>
            <span>{timePassed && username}</span>
        </div>
    )
}