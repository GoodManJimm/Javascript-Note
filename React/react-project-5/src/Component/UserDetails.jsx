import PropTypes from "prop-types";
import LoginForm from "./LoginForm";
import {useState} from 'react'

export function UserDetails({user, setUsers})
{
    const [isEditing,setIsEditing] = useState(false)
    const [userName,setUsername] = useState(user.username)
    const [email,setEmail] = useState(user.email)

    console.log(isEditing);
    return (<div data-testid={'user-details-' + user.id } key={user.id}>
        <button data-testid={'edit-btn-' + user.id } onClick={() => 
            {
                setIsEditing((currentState) => !currentState)
            }}>Edit</button>

        <button onClick={() => setUsers((currentUsersState) => currentUsersState.filter(
            (currentUser) => currentUser.id !== user.id
            ))}>Delete</button>

        { isEditing && <button onClick={() => 
        {
            setUsers((currentUsersState) => 
                {
                    return currentUsersState.map((currentUser) => 
                        (currentUser.id === user.id) ? {...currentUser,username : userName,email : email} : currentUser
                        )
                })
            setIsEditing(false)
        }
        }>save</button> }

        <br />
        <b>ID: </b>
        <span>{user.id}</span>
        <br />
        {isEditing ? <label htmlFor="username">Username:</label> : <b>userName: </b>}
        {isEditing ? <input aria-label="username" name="username" id="username" value={userName} onChange={
            (e) => 
                {
                    setUsername(e.target.value)
                }
        }/> : <span> {user.username} </span>}
        <br />
        {isEditing ? <label htmlFor="email">Email:</label>: <b>Email: </b>}
        {isEditing ? <input aria-label="email" name="email" id="email" value={email} onChange=
        {
            (e) => 
                {
                    setEmail(e.target.value)
                }
        }/> : <span> {user.email} </span>}
      </div>
)
}

UserDetails.propTypes
{
    user: PropTypes.shape(
        {
            id: PropTypes.number.isRequired,
            username: PropTypes.string.isRequired,
            email: PropTypes.string.isRequired
        }).isRequired
    setUsers: PropTypes.func.isRequired
}