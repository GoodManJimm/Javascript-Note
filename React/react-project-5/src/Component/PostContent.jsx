import { useContext } from "react";
import { UserContext } from "../Context/UserContext";
import PostContentButton from "./PostContentButton";


export function PostContent()
{
    const userContextData = useContext(UserContext)

    return (
        <div>
            <div>
                <h3>Post Content</h3>
                <br />
                <div>
                    {userContextData.email}
                    <PostContentButton/>
                </div>
            </div>
        </div>
    )
}