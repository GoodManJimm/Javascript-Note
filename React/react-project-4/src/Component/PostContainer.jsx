import { useContext } from "react"
import {PostContent} from "./PostContent"
import { UserContext } from "../Context/UserContext"

export function PostContainer()
{
    const userContextData = useContext(UserContext)

    // console.log(userContextData);
    return (
        <div>
            <div>
                <span>Container</span>
            </div>
            <div>{userContextData.name}</div>
            <div>{userContextData.id}</div>
            <div>{userContextData.email}</div>
            <div>{userContextData.username}</div>
            <br />
        </div>
    )
}