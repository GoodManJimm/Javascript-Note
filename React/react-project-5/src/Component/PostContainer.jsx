import { useContext } from "react"
import {PostContent} from "./PostContent"
import { UserContext } from "../Context/UserContext"

export function PostContainer()
{
    const userContextData = useContext(UserContext)

    // console.log(userContextData);
    return (
        <div>
            <div>
                <span>Container</span>
            </div>
            <div>Display Name:{userContextData.name}</div>
            <div>ID: {userContextData.id}</div>
            <div>Email: {userContextData.email}</div>
            <div>Username: {userContextData.username}</div>
            <PostContent/>
            <br />
        </div>
    )
}