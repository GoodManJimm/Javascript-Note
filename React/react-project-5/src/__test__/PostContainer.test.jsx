import { expect,it, describe } from "vitest";
import { screen,render } from "@testing-library/react";
import { PostContainer } from "../Component/PostContainer";
import { UserContext } from "../Context/UserContext";

describe("render context values", () => {
    it ('should match snap shot', () => {
        const mockUserContextData = {
            id: 1001,
            username: 'johnny',
            email: 'johnny@gmail.com',
            name: 'johnny',
            setUserData: () => {}
        }
        const {container} = render(
            <UserContext.Provider value={{...mockUserContextData}}>
                <PostContainer />
            </UserContext.Provider>)
            expect(container).toMatchSnapshot()
    })

    it("should display username", () => {
        const mockUserContextData = {
            id: 1001,
            username: 'johnny',
            email: 'johnny@gmail.com',
            name: 'johnny',
            setUserData: () => {}
        }
        const {container} = render(
        <UserContext.Provider value={{...mockUserContextData}}>
            <PostContainer />
        </UserContext.Provider>)
        expect(container).toMatchSnapshot()
        expect(screen.getByText("Username: johnny")).toBeInTheDocument()
    })

})