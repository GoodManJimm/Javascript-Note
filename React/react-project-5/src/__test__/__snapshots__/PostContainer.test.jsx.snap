// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render context values > should display username 1`] = `
<div>
  <div>
    <div>
      <span>
        Container
      </span>
    </div>
    <div>
      Display Name:
      johnny
    </div>
    <div>
      ID: 
      1001
    </div>
    <div>
      Email: 
      johnny@gmail.com
    </div>
    <div>
      Username: 
      johnny
    </div>
    <div>
      <div>
        <h3>
          Post Content
        </h3>
        <br />
        <div>
          johnny@gmail.com
          <div>
            <div>
              <span>
                PostContentButton
              </span>
            </div>
            1001
            <br />
            <label
              for="UpdateName"
            >
              Update Name:
            </label>
            <input
              id="UpdateName"
              type="text"
              value=""
            />
            <button>
              Update display name
            </button>
          </div>
        </div>
      </div>
    </div>
    <br />
  </div>
</div>
`;

exports[`render context values > should match snap shot 1`] = `
<div>
  <div>
    <div>
      <span>
        Container
      </span>
    </div>
    <div>
      Display Name:
      johnny
    </div>
    <div>
      ID: 
      1001
    </div>
    <div>
      Email: 
      johnny@gmail.com
    </div>
    <div>
      Username: 
      johnny
    </div>
    <div>
      <div>
        <h3>
          Post Content
        </h3>
        <br />
        <div>
          johnny@gmail.com
          <div>
            <div>
              <span>
                PostContentButton
              </span>
            </div>
            1001
            <br />
            <label
              for="UpdateName"
            >
              Update Name:
            </label>
            <input
              id="UpdateName"
              type="text"
              value=""
            />
            <button>
              Update display name
            </button>
          </div>
        </div>
      </div>
    </div>
    <br />
  </div>
</div>
`;
